tests/helpers/integration_setup.py

```diff
@@ -40,11 +39,15 @@ def stop_service():
 
     if container_process is not None:
         logger.info('Stopping container')
-        container_process.send_signal(signal.SIGTERM)
+
+        # Stop and remove containers
+        cwd = 'tests/integration/docker'
+        subprocess.run("docker compose --ansi never down", shell=True, check=True, cwd=cwd)
+
         logger.info('Waiting until service has stopped...')
 
-        if not common.wait_for_line("container.err",
-                                    lambda line: 'Stopping' in line and 'done' in line,
+        if not common.wait_for_line("container.out",
+                                    lambda line: "exited with code 0" in line,
                                     timeout=stop_timeout,
                                     line_count=1):
             raise Exception(f'Container did not stop in the alloted time of {stop_timeout} seconds')
```

Comment:
Since subprocess run will block until docker compose down is complete, I don't think all this checking code does anything anymore. It won't run until the down is finished

Reply:
I believe this check is useful and necessary because it verifies that the containers are actually shutting down.

Reply:
They must be down at this point. Run blocks until the command exits, and check ensures that the exit code is 0. If not, an exception will be thrown. This code can only be reached if the containers are down

Reply:
fixed

Reply:
Interestingly the tests are no longer passing. I don't see how the changes you made could've caused that...

Reply:
![Uploading Screenshot 2025-05-16 at 5.49.02 PM.png…]()

Reply:
<img width="839" alt="Screenshot 2025-05-16 at 5 49 02 PM" src="https://github.com/user-attachments/assets/975c413c-0472-4359-9c9c-4389d10e8807" />

Reply:
removed .common